### PR TITLE
Fix #11 Add context object with get and run to the javabridge object

### LINKED ISSUE #11: Add context object with get and run to the javabridge object

### PR DESCRIPTION
**Pull request by ComLock**

### CODE REVIEW

File: src/context.ts [hunk 1]
```diff
@@ -1,40 +1,133 @@
...
-				idProvider: contextParams.user.idProvider || undefined,
-			} as User : null,
-			principals: contextParams.principals || null,
-		};
+			principals: contextParams.principals
+		}
+	}
+
+	if (contextParams.user) {
+		if (contextParams.user.login) {
```
Comment (alansemenov) [inline src/context.ts:53]:
```suggestion
	if (contextParams.user?.login?) {
		if (contextParams.user.login) {
```

File: test/content/create.test.ts [hunk 2]
```diff
@@ -1,72 +1,85 @@
...
-			id: 'com.enonic.cms.default'
+xp.repo.createBranch({
+    branchId: 'draft',
+    repoId: REPO
+});
+
+
+describe('content', () => {
+	describe('create', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/create.test.ts:39]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/createMedia.test.ts [hunk 3]
```diff
@@ -1,102 +1,117 @@
...
+
+xp.repo.createBranch({
+    branchId: 'draft',
+    repoId: REPO
+});
+
+
+describe('content', () => {
+	describe('createMedia', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/createMedia.test.ts:45]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/delete.test.ts [hunk 4]
```diff
@@ -1,66 +1,71 @@
...
+		contentType: 'base:folder',
+		data: {},
+		name: 'folder',
+		parentPath: '/',
+	});
+});
+
+describe('content', () => {
+	describe('delete', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/delete.test.ts:53]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/exists.test.ts [hunk 5]
```diff
@@ -1,63 +1,85 @@
...
-		const javaBridge = new JavaBridge({
-			app: {
-				config: {},
-				name: APP_NAME,
-				version: '0.0.1-SNAPSHOT'
-			},
-			log
+describe('content', () => {
+	describe('exists', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/exists.test.ts:52]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/exists.test.ts [hunk 6]
```diff
@@ -1,63 +1,85 @@
...
-					});
-					it('returns true when key is a path and content exists', () => {
-						expect(contentConnection.exists({ key: createdContent._path })).toBe(true);
-					});
-					it("returns false when content doesn't exist", () => {
-						expect(contentConnection.exists({ key: 'non-existant-id' })).toBe(false);
-						expect(contentConnection.exists({ key: '/non-existant-path' })).toBe(false);
-					});
+
+		it('returns false for an non-existant content', () => {
```
Comment (alansemenov) [inline test/content/exists.test.ts:72]:
```suggestion
		it('returns false for a non-existent content', () => {
```

File: test/content/exists.test.ts [hunk 7]
```diff
@@ -1,63 +1,85 @@
...
+			const fn = () => {
+				return run({
+					currentApplicationKey: APP,
+					branch: 'draft',
+					repository: REPO,
+				},() => {
+					return exists({ key: 'non-existant' });
```
Comment (alansemenov) [inline test/content/exists.test.ts:79]:
```suggestion
					return exists({ key: 'non-existent' });
```

File: test/content/get.test.ts [hunk 8]
```diff
@@ -1,64 +1,88 @@
...
+		contentType: 'base:folder',
+		data: {},
+		name: 'folder',
+		parentPath: '/',
+	});
+});
+
+describe('content', () => {
+	describe('get', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/get.test.ts:52]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/getAttachmentStream.test.ts [hunk 9]
```diff
@@ -1,72 +1,93 @@
...
+		mimeType: 'image/jpeg',
+		focalX: 0.5,
+		focalY: 0.5,
+	});
+});
+
+
+describe('content', () => {
+	describe('getAttachmentStream', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/getAttachmentStream.test.ts:63]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/modify.test.ts [hunk 10]
```diff
@@ -1,100 +1,110 @@
...
+		contentType: 'base:folder',
+		data: {},
+		name: 'folder',
+		parentPath: '/',
+	});
+});
+
+describe('content', () => {
+	describe('modify', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/modify.test.ts:52]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/move.test.ts [hunk 11]
```diff
@@ -1,102 +1,96 @@
...
-							source: createdContent._id,
-							target
-						});
-						// log.debug('movedContent:%s', movedContent);
-						expect(movedContent._name).toBe(target);
-						expect(movedContent._path).toBe('/anotherNewPath/newName'); // just new name
-					});
+describe('content', () => {
+	describe('move', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/content/move.test.ts:52]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/content/create.test.ts (hunk 2 above)
Comment (ComLock) [reply test/content/create.test.ts]: Fixed

File: test/content/delete.test.ts (hunk 4 above)
Comment (ComLock) [reply test/content/delete.test.ts]: Fixed

File: test/portal/imageUrl.test.ts [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+					return imageUrl({
+						id: leaSeydouxJpg._id,
+						scale: 'block(1920,1280)',
+					});
+				});
+			}
+			expect(fn()).toEqual('/admin/site/preview/myproject/draft/some/path/_/image/00000000-0000-4000-8000-000000000004:0123456789abcdef/block-1920-1280/Lea-Seydoux.jpg');
+		});
+
+		it('should handle preview by path inside Content Studio', () => {
```
Comment (alansemenov) [inline test/portal/imageUrl.test.ts:172]:
```suggestion
		it('should handle preview by path under admin', () => {
```

File: test/portal/imageUrl.test.ts [hunk 13]
```diff
@@ -0,0 +1,215 @@
...
+						quality: 100,
+						scale: 'full',
+						type: 'absolute'
+					});
+				});
+			}
+			expect(fn()).toEqual('http://localhost:8080/admin/site/preview/myproject/draft/some/path/_/image/00000000-0000-4000-8000-000000000004:0123456789abcdef/full/Lea-Seydoux.jpg.png?key=value&background=FFFFFF&filter=rounded%285%29%3Bsharpen%28%29&quality=100');
+		});
+
+		it('should handle preview by id outside Content Studio', () => {
```
Comment (alansemenov) [inline test/portal/imageUrl.test.ts:197]:
```suggestion
		it('should handle preview by id outside admin', () => {
```

File: test/portal/getContent.test.ts [hunk 14]
```diff
@@ -0,0 +1,78 @@
...
+const aContent = contentConnection.create({
+	contentType: 'myContentType',
+	data: {},
+	name: 'myContent',
+	parentPath: '/',
+});
+
+describe('portal', () => {
+	describe('getContent', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/portal/getContent.test.ts:39]:
```suggestion
		it('should throw when there is no context', () => {
```

File: test/portal/assetUrl.test.ts [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+						params: {
+							key: 'value'
+						},
+					});
+				});
+			}
+			expect(fn()).toEqual('/site/myproject/master/some/path/_/asset/com.enonic.app.myapp1:0123456789abcdef//images/asset.png?key=value');
+		});
+
+		it('should handle live url outside Content Studio with type absolute', () => {
```
Comment (alansemenov) [inline test/portal/assetUrl.test.ts:163]:
```suggestion
		it('should handle live url with type absolute outside admin', () => {
```

File: test/portal/assetUrl.test.ts [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+							key: 'value'
+						},
+						type: 'absolute'
+					});
+				});
+			}
+			expect(fn()).toEqual('http://localhost:8080/admin/site/preview/myproject/draft/some/path/_/asset/com.enonic.app.myapp2:0123456789abcdef//images/asset.png?key=value');
+		});
+
+		it('should handle live url outside Content Studio', () => {
```
Comment (alansemenov) [inline test/portal/assetUrl.test.ts:144]:
```suggestion
		it('should handle live url outside admin', () => {
```

File: test/portal/assetUrl.test.ts [hunk 17]
```diff
@@ -0,0 +1,184 @@
...
+						params: {
+							key: 'value'
+						},
+					});
+				});
+			}
+			expect(fn()).toEqual('/admin/site/preview/myproject/draft/some/path/_/asset/com.enonic.app.myapp1:0123456789abcdef//images/asset.png?key=value');
+		});
+
+		it('should handle preview inside Content Studio with type absolute', () => {
```
Comment (alansemenov) [inline test/portal/assetUrl.test.ts:123]:
```suggestion
		it('should handle preview with type absolute under admin', () => {
```

File: test/portal/assetUrl.test.ts [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+							key: 'value'
+						},
+						type: 'absolute'
+					});
+				});
+			}
+			expect(fn).toThrow(/mock-xp: lib-portal.assetUrl\(\): No context.request\! Support for assetUrl outside portal is not yet implemented./);
+		});
+
+		it('should handle preview inside Content Studio', () => {
```
Comment (alansemenov) [inline test/portal/assetUrl.test.ts:104]:
```suggestion
		it('should handle preview under admin', () => {
```

File: test/portal/assetUrl.test.ts [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+	log
+});
+
+const run = xp.context.run;
+const assetUrl = xp.portal.assetUrl;
+
+
+describe('portal', () => {
+	describe('assetUrl', () => {
+		it('should throw when there is context is not found', () => {
```
Comment (alansemenov) [inline test/portal/assetUrl.test.ts:70]:
```suggestion
		it('should throw when there is no context', () => {
```

**Review by alansemenov — APPROVED** [review]
Great stuff

**Pull request merged**